staging/src/k8s.io/apiserver/pkg/server/config.go

```diff
@@ -328,7 +330,14 @@ func NewConfig(codecs serializer.CodecFactory) *Config {
 	defaultHealthChecks := []healthz.HealthChecker{healthz.PingHealthz, healthz.LogHealthz}
 	var id string
 	if utilfeature.DefaultFeatureGate.Enabled(genericfeatures.APIServerIdentity) {
-		id = "kube-apiserver-" + uuid.New().String()
+		hostname, err := os.Hostname()
+		if err != nil {
+			klog.Fatalf("error getting hostname for apiserver identity: %v", err)
+		}
+
+		h := fnv.New32a()
```

Comment:
I would prefer a real hash function, this is only called at startup, there's no performance problem. Can you switch to SHA256? It is OK to truncate the result a bit, but 4 bytes is too small. 

If we assume there are 1M k8s clusters, each with 3 apiservers, there is a `1000000*(1-e^(-(3^2)/(2^32))) = .2%` one of them will have a collision. That's far too high, I would like a chance less than 1e-9. (google "birthday paradox estimator" if you want to check my math)

6 bytes is enough to pass that test, so I would go to 8 bytes selected from SHA256, where there is no question that it is very random.

Reply:
(and sorry I wasn't able to review this until now)

Reply:
(oh and if you want you can use the base58 encoder to save some characters)

Reply:
Why bother truncating at all?

Reply:
Opened https://github.com/kubernetes/kubernetes/pull/113649 -- tests seem fine without needing to truncate, but still might be worthwhile just to avoid lease names that are really long. I don't feel too strongly with the lease name being that long but myabe some people do?